ui/pages/confirmations/components/confirm/splash/smart-account-update/smart-account-update.tsx

```diff
@@ -70,25 +79,32 @@ const ListItem = ({
 export function SmartAccountUpdate() {
   const [acknowledged, setAcknowledged] = useState(false);
   const t = useI18nContext();
+  const dispatch = useDispatch();
   const { currentConfirmation } = useConfirmContext<TransactionMeta>();
   const { handleRejectUpgrade } = useSmartAccountActions();
   const splashPageAcknowledgedForAccountList: string[] = useSelector(
     getUpgradeSplashPageAcknowledgedForAccounts,
   );
-
+  const smartAccountOptIn = useSelector(getUseSmartAccount);
   const { chainId, txParams, origin } = currentConfirmation ?? {};
   const { from } = txParams;
+  const account = useSelector((state: AccountsState) =>
+    getMemoizedInternalAccountByAddress(state as AccountsState, from),
+  );
+  const keyringType = account?.metadata?.keyring?.type;
 
   const acknowledgeSmartAccountUpgrade = useCallback(() => {
     setSplashPageAcknowledgedForAccount(from);
     setAcknowledged(true);
+    dispatch(setSmartAccountOptIn(true));
   }, [from, setAcknowledged]);
 
   if (
     !currentConfirmation ||
     acknowledged ||
     origin === ORIGIN_METAMASK ||
-    splashPageAcknowledgedForAccountList.includes(from.toLowerCase())
+    splashPageAcknowledgedForAccountList.includes(from.toLowerCase()) ||
+    (smartAccountOptIn && keyringType === KeyringTypes.hd)
```

Comment:
See comment above, but for reference we also support `simple`.

See `SUPPORTED_KEYRING_TYPES` in `eip5792.ts`.

Reply:
We want to prompt user for imported accounts which have `simple` type.